databases/functions/userFollowUser/check-is-blurred-post.js

```diff
@@ -0,0 +1,25 @@
+const {MINIMUM_BLUR_USER_FOLLOWING} = require('../../../helpers/constants');
+
+/**
+ *
+ * @param {Model} userFollowUserModel
+ * @param {RegisterBodyData.Users} users
+ */
+module.exports = async (userFollowUserModel, userIdFollower, _transaction = null) => {
+  let isBlurredPost = false;
+
+  const signUserList = await userFollowUserModel.findAll({
+    where: {
+      user_id_follower: userIdFollower,
+      is_anonymous: false
+    },
+    limit: MINIMUM_BLUR_USER_FOLLOWING,
+    raw: true
+  });
+
+  if (signUserList.length <= MINIMUM_BLUR_USER_FOLLOWING) {
+    isBlurredPost = true;
+  }
+
+  return isBlurredPost;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in this function seems to be incorrect. You are fetching the followers of a user up to `MINIMUM_BLUR_USER_FOLLOWING` and then checking if the number of followers is less than or equal to `MINIMUM_BLUR_USER_FOLLOWING`. This will always return true because you are limiting the number of followers fetched to `MINIMUM_BLUR_USER_FOLLOWING`.

You should fetch all followers and then check if their count is less than `MINIMUM_BLUR_USER_FOLLOWING`. Also, consider renaming `MINIMUM_BLUR_USER_FOLLOWING` to something like `BLUR_THRESHOLD` for better readability.

```diff
-   const signUserList = await userFollowUserModel.findAll({
-     where: {
-       user_id_follower: userIdFollower,
-       is_anonymous: false
-     },
-     limit: MINIMUM_BLUR_USER_FOLLOWING,
-     raw: true
-   });
+   const signUserList = await userFollowUserModel.count({
+     where: {
+       user_id_follower: userIdFollower,
+       is_anonymous: false
+     }
+   });

-   if (signUserList.length <= MINIMUM_BLUR_USER_FOLLOWING) {
+   if (signUserList < BLUR_THRESHOLD) {
      isBlurredPost = true;
    }
```
Please note that I've used the `count` method instead of `findAll` to get the total number of non-anonymous followers. This is more efficient as it avoids fetching all the data when we only need the count.


<!-- This is an auto-generated comment by OSS CodeRabbit -->